doc/releases/migration-guide-3.5.rst

```diff
@@ -271,6 +271,17 @@ Required changes
 * The :kconfig:option:`CONFIG_RISCV_MTVEC_VECTORED_MODE` Kconfig option was renamed to
   :kconfig:option:`CONFIG_RISCV_VECTORED_MODE`.
 
+* ZBus runtime observers implementation now relies on the HEAP memory instead of a memory slab.
+  Thus, zbus' configuration (kconfig) related to runtime observers has changed. To keep your runtime
+  observers code working correctly, you need to:
+
+  - Replace the integer :kconfig:option:`CONFIG_ZBUS_RUNTIME_OBSERVERS_POOL_SIZE` with the boolean
```

Comment:
nit: it probably doesn't make sense to use :kconfig: to point to an option that doesn't exist anymore

```suggestion
  - Replace the integer ``CONFIG_ZBUS_RUNTIME_OBSERVERS_POOL_SIZE`` with the boolean
```